tests/unit/Framework/MockObject/Builder/InvocationMockerTest.php

```diff
@@ -243,4 +244,37 @@ public function testWillReturnAlreadyInstantiatedStubs(): void
         $this->assertSame('foo', $mock->foo());
         $this->assertSame($mock, $mock->bar());
     }
+
+    public function testConsecutiveCallsInOrder(): void
+    {
+        $mock = $this->createMock(ClassWithVariadicArgumentMethod::class);
+        $mock->expects($this->exactly(2))
+            ->method('foo')
+            ->with(1)
+            ->willReturn('first')
+            ->andThen($this->once())
+            ->with(2)
+            ->willReturn('second');
```

Comment:
This is a bit weird, "strict" call-order is not possible when same arguments are given multiple times.
I would have expected, this would work as well, but it fails:

$mock->expects($this->once())
            ->method('foo')
            ->with(1)
            ->willReturn('first')
            ->andThen($this->once())
            ->with(2)
            ->willReturn('second')
            ->andThen($this->once())
            ->with(1)
            ->willReturn('first');

Reply:
In general, I like the idea of having a substitution of withConsecutive() using the fluent interface. The details are a bit nifty though, see my example above. It's pretty hard to end up with a simple syntax without surprises. andThen() suggests a "strict" order, while "orThen" is "may be called with this as well, and if so, then return that". That's not entirely clear to me with proposed solution.